utils/cryptography/encryption.go

```diff
@@ -0,0 +1,54 @@
+package crytography
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"io"
+	"os"
+
+	"golang.org/x/crypto/nacl/secretbox"
+)
+
+// EncryptionKey for encrypting secrets
+var EncryptionKey, _ = base64.StdEncoding.DecodeString(os.Getenv("ENCRYPTION_KEY"))
+
+// Both what the encryption key length should be
+var keyLength = 32
+
+// EncryptMessage uses SecretBox to encrypt the message
+func EncryptMessage(field []byte) (encrypted []byte, nonceString [24]byte, err error) {
+	var nonce [24]byte
+
+	// The key argument should be 32 bytes long
+	if len(EncryptionKey) != keyLength {
+		return nil, nonce, errors.New("Encryption Key is not the correct key length")
+	}
+
+	var encryptionKey [32]byte
+	copy(encryptionKey[:], []byte(EncryptionKey))
+
+	if _, err := io.ReadFull(rand.Reader, nonce[:]); err != nil {
+		return nil, nonce, err
+	}
+
+	var out []byte
+	encryptedField := secretbox.Seal(out[:], field, &nonce, &encryptionKey)
+
+	return encryptedField, nonce, nil
+}
+
+// DecryptMessage uses SecretBox to decrypt the message
+func DecryptMessage(encryptedField []byte, nonce []byte) (string, error) {
+	var encryptionKey [32]byte
+	copy(encryptionKey[:], []byte(EncryptionKey))
```

Comment:
is this timing-safe? i don't see why it wouldn't be but just checking. @evq

Reply:
actually i guess this doesn't matter since attacker can't easily control the message being decrypted

Reply:
right, or the key that's being copied here? ( wasn't sure if your comment was on the copy in particular )

Reply:
perhaps we should just be doing this copy once though in an `init` function

Reply:
Reducing footprint by running `Init` once :) https://github.com/brave-intl/bat-go/pull/347/commits/f5360b043f9b493e96baffc3d1d327aa1dfae20a
https://github.com/brave-intl/bat-go/pull/347/commits/5d98abf5e0c95c9e25612c9624217bd842ab6cb9